libs/stocks/feature-shell/src/lib/stocks/stocks.component.ts

```diff
@@ -11,33 +12,26 @@ export class StocksComponent implements OnInit {
   stockPickerForm: FormGroup;
   symbol: string;
   period: string;
-
+  maxDate = new Date();
   quotes$ = this.priceQuery.priceQueries$;
 
-  timePeriods = [
-    { viewValue: 'All available data', value: 'max' },
-    { viewValue: 'Five years', value: '5y' },
-    { viewValue: 'Two years', value: '2y' },
-    { viewValue: 'One year', value: '1y' },
-    { viewValue: 'Year-to-date', value: 'ytd' },
-    { viewValue: 'Six months', value: '6m' },
-    { viewValue: 'Three months', value: '3m' },
-    { viewValue: 'One month', value: '1m' }
-  ];
-
   constructor(private fb: FormBuilder, private priceQuery: PriceQueryFacade) {
     this.stockPickerForm = fb.group({
       symbol: [null, Validators.required],
-      period: [null, Validators.required]
-    });
+      fromDate: [null, [Validators.required, CustomDateValidator.fromToDateValidator]],
+      toDate: [null, [Validators.required, CustomDateValidator.fromToDateValidator]]
+    },
+      { validator: CustomDateValidator.fromToDateValidator('fromDate', 'toDate') }
+    );
   }
 
   ngOnInit() {}
 
   fetchQuote() {
     if (this.stockPickerForm.valid) {
-      const { symbol, period } = this.stockPickerForm.value;
-      this.priceQuery.fetchQuote(symbol, period);
+      const { symbol, fromDate, toDate } = this.stockPickerForm.value;
+      const period = 'max';
```

Comment:
Using max as period for all the from and to date request and filtering the data whenever we receive response from the API.